wordpress-mobile/WordPress-Android
# Preserve site multi-selections in site picker on configuration change.

Fixes #8350.

## To test:

0. Use an account with at least 3 WordPress.com sites, start in portrait mode
1. select Switch Site
2. Make sure at least one WordPress.com site is hidden. To hide a site, long press on any site on the list until the toolbar changes. Select a site, then select "Hide" at the top. The site should be hidden from the list.
3. Now the actual testing can start. Long press on any  site in the list to open the multiple editing mode again. Doing this should make the hidden site appear.
4. Select a visible site. This should enable the "Hide" menu at the top.
5. Turn device to landscape mode. This should trigger layout change. Make sure that the selected site remains selected, and the "Hide" menu remains enabled.
6. Switch back to portrait mode.
7. De-select the visible site. Select a hidden site. This should enable the "Show" menu at the top.
8. Turn device to landscape mode. This should trigger layout change. The selected site should remain selected, and the "Show" menu should remain enabled.
9. Switch back to portrait mode.
10. De-select the hidden site. Select a visible site and a hidden site. This should enable both "Show" and "Hide" menu at the top.
11. Turn device to landscape mode. This should trigger layout change. The selected sites should remain selected, and the "Show" and "Hide" menu items should remain enabled.
12. Switch back to portrait mode.
13. Play around with using the Show/Hide/Select All/Deselect All menu options. The functionalities should remain working and unaffected by this PR.

## More info:
This change involves more moving parts than I initially thought, so I wrote down explanation of what the code change did [on this Gist](https://gist.github.com/hafizrahman/6c1e829a52d7714c540b379397e52d09).

## PR submission checklist:
